src/Compatibility/Core/src/Android/ResourceManager.cs

```diff
@@ -23,23 +23,39 @@ public static class ResourceManager
 		static ImageCache GetCache() => _lruCache.Value;
 
 		static Assembly _assembly;
+		[UnconditionalSuppressMessage ("Trimming", "IL2026", Justification = "Resource.designer.cs is in the root application assembly, which should be preserved.")]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.PublicProperties)]
```

Comment:
I think it's worth at least creating a bug for this. It's true that by default this is going to be the case (at least I think), but it's still user-configurable, so it is possible to get to the situation where it could be trimmed away.

Since it seems we know statically what types we want from that assembly, this might be solvable - either via source generation or some other mechanism.

Reply:
I think this one is OK.

This class is reflecting on a type that the Android build system generates: `Resource.designer.cs`

It is a class with a bunch of `static readonly int` fields that represent Android resource files. The way it is implemented, the main app assembly (should be the root) always contains the class and it is never linked away. It is the C# implementation of the Android [`R.java`](https://www.javatpoint.com/R-java-file-in-android) file.

Reply:
If I change the settings for the app to run the main assembly with "link", would it not trim away even this type (or the members in it)?